src/main/java/frc/robot/subsystems/ShooterSubsystem.java

```diff
@@ -36,4 +49,9 @@ public void shoot(double shooterSpeed) {
         // shooterController.setReference(shooterSpeed, ControlType.kVelocity);
         shooterMotor1.set(shooterSpeed);
     }
+
+    @Override
+    public void periodic() {
+        rangeFilter.calculate(ballSensor.getRangeInches());
```

Comment:
What's the point of calculating this every loop if you calculate it again when you call `isBallReady`

Reply:
Calculating it in each loop makes sure that the `rangeFiler` has an up-to-date distance value, as it takes a few `calculate` calls for it to update to the true value. I would actually just not call `calculate` in `isBallReady` in favor of some `getFilteredValue` method, but I don't think that exists.

Reply:
Oh I see. Ok gotcha